parts/linux/cloud-init/artifacts/cse_config.sh

```diff
@@ -551,7 +551,15 @@ ensureGPUDrivers() {
     fi
 
     if [[ "${CONFIG_GPU_DRIVER_IF_NEEDED}" = true ]]; then
-        logs_to_events "AKS.CSE.ensureGPUDrivers.configGPUDrivers" configGPUDrivers
+        existing_version="$(nvidia-smi | grep "Driver Version" | cut -d' ' -f3)"
+
+        if [[ "*$existing_version*" == $NVIDIA_DRIVER_IMAGE_TAG ]]; then
```

Comment:
I think this will confuse cuda/grid drivers of the same version which are incompatible